examples/dense_bubble.py

```diff
@@ -75,8 +80,13 @@
     zc = 3000.
     zr = 2000.
     r = sqrt(((x[0]-xc)/xr)**2 + ((x[1]-zc)/zr)**2)
-    theta_pert = conditional(r > 1., 0., -7.5*(1.+cos(pi*r)))
-    theta0.interpolate(theta_b + theta_pert)
+
+    physics_boundary = Boundary_Method.physics if recovered else None
+    rho_b_Vt = Function(Vt)
+    rho_b_recoverer = Recoverer(rho_b, rho_b_Vt, VDG=Vt_brok, boundary_method=physics_boundary).project()
```

Comment:
I actually called `project` in the same line as declaring the `Recoverer` which was a bit lazy. I'll separate them out to make it a bit clearer.

It's necessary for both sets of spaces as it is used to take `rho` into `Vt`, so that we algebraically evaluate exner pi in `Vt`